File: app/src/main/java/otang/news/api/ApiInterface.java
package otang.news.api;

import otang.news.model.NewsResponse;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface ApiInterface {
	@GET("/v2/top-headlines")
	Call<NewsResponse> getTopHeadline(@Query("country") String country, @Query("apiKey") String apikey);

	@GET("/v2/top-headlines")
	Call<NewsResponse> getByCategory(@Query("category") String category, @Query("country") String country,
			@Query("apiKey") String apikey);

	@GET("/v2/everything")
	Call<NewsResponse> findNews(@Query("q") String keyword, @Query("apiKey") String apikey);
}

File: app/src/main/java/otang/app/news/api/ApiInterface.kt
package otang.app.news.api

import otang.app.news.model.NewsResponse
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Query

interface ApiInterface {
    @GET("/v2/top-headlines")
    fun getTopHeadline(
        @Query("country") country: String?,
        @Query("apiKey") apikey: String?
    ): Call<NewsResponse?>

    @GET("/v2/top-headlines")
    fun getByCategory(
        @Query("category") category: String?, @Query("country") country: String?,
        @Query("apiKey") apikey: String?
    ): Call<NewsResponse?>

    @GET("/v2/everything")
    fun findNews(
        @Query("q") keyword: String?,
        @Query("apiKey") apikey: String?
    ): Call<NewsResponse?>
}